includes/class-simple-local-avatars.php

```diff
@@ -678,4 +716,158 @@ public function get_avatar_rest( $user ) {
+			$migrations      = get_option( 'simple_local_avatars_migrations', array() );
+			$processed_users = isset( $migrations['wp_user_avatar'] ) ? $migrations['wp_user_avatar'] : array();
+
+			// Get all users that have a local avatar.
+			$users = get_users(
+				array(
+					'blog_id'      => $blog_id,
+					'exclude'      => $processed_users,
+					'meta_key'     => $meta_key,
+					'meta_compare' => 'EXISTS',
+				)
+			);
+
+			// Bail early if we don't find users.
+			if ( empty( $users ) ) {
+				continue;
+			}
+
+			foreach ( $users as $user ) {
+				// Get the existing avatar media ID.
+				$avatar_id = get_user_meta( $user->ID, $meta_key, true );
+
+				// Attach the user and media to Simple Local Avatars.
+				$sla = new Simple_Local_Avatars();
+				$sla->assign_new_user_avatar( (int) $avatar_id, $user->ID );
+
+				// Check that it worked.
+				$is_migrated = get_user_meta( $user->ID, 'simple_local_avatar', true );
+
+				if ( ! empty( $is_migrated ) ) {
+					// Build array of user IDs.
+					$migrations['wp_user_avatar'][] = $user->ID;
+
+					// Record the user IDs so we don't process a second time.
+					$is_saved = update_option( 'simple_local_avatars_migrations', $migrations );
+
+					// Record how many avatars we migrate to be used in our messaging.
+					if ( $is_saved ) {
+						$count++;
+					}
+				}
+			}
+		}
+
+		return $count;
+
+	}
+
+	/**
+	 * Migrate the user's avatar data away from WP User Avatar/ProfilePress via the dashboard.
+	 *
+	 * Sends the number of avatars processed back to the AJAX response before stopping execution.
+	 *
+	 * @return void
+	 */
+	public function ajax_migrate_from_wp_user_avatar() {
+		// Bail early if nonce is not available.
+		if ( empty( $_POST['migrateFromWpUserAvatarNonce'] ) ) {
+			die;
+		}
+
+		// Bail early if nonce is invalid.
+		if ( ! wp_verify_nonce( $_POST['migrateFromWpUserAvatarNonce'], 'migrate_from_wp_user_avatar_nonce' ) ) {
+			die();
+		}
+
+		// Run the migration script and store the number of avatars processed.
+		$count = $this->migrate_from_wp_user_avatar();
+
+		// Create the array we send back to javascript here.
+		$array_we_send_back = array( 'count' => $count );
+
+		// Make sure to json encode the output because that's what it is expecting.
+		echo wp_json_encode( $array_we_send_back );
+
+		// Make sure you die when finished doing ajax output.
+		wp_die();
+
+	}
+
+	/**
+	 * Migrate the user's avatar data from WP User Avatar/ProfilePress via the command line.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--yes]
+	 * : Skips the confirmations (for automated systems).
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp simple-local-avatars migrate wp-user-avatar
+	 *     Success: Number of avatars successfully migrated from WP User Avatar: 5
+	 *
+	 * @param array|null $args       The arguments.
+	 * @param array|null $assoc_args The associative arguments.
+	 *
+	 * @return void
+	 */
+	public function wp_cli_migrate_from_wp_user_avatar( $args = null, $assoc_args = null ) {
```

Comment:
Is it required to set default values?

Reply:
Resolved by removing the default values.